docs/common/coffee-model.adoc

```diff
@@ -11,9 +11,60 @@ Ezekre nyújt ősosztályokat, id generátort és audit mező töltőt.
 Tartamazza a deltaspike DATA bekötését, mely a criteria API könnyítését segíti,
 és a maven ezekre az ősökre kigenerálja a SingularAttribute osztályokat is.
 
+=== ID generálás
+Az ID generálás célja, hogy minden körülmény között egyedi és nem sorfolytonos ID értékekkel (string) dolgozhassunk.footnote:[Milyért string tipust használunk az azonosítokra és nem int-et? *Biztonsági szempontból*, az azonosítoknak nem szabad sorfolytonosnak lenniük, különben könnyen kitalálhatóak, *hordozhatóság szempontjából* egy esetleges db migráció során az int tipusú azonosítok könnyen összeakadhatnak a már meglévő adatokkal, *bővíthetőség szempontjából* nagyobb a mozgásterünk, *mennyiség szempontjából* nem vagyunk lekorlátozva, *hordozott információ szempontjából* nem derül ki a rendszerről az adatkör mennyisége.] A generáláshoz az `EntityIdGenerator.java` osztályt használjuk, ez alapértelmezetten az entity java osztályon annotávióval van használva, de ha szükség van rá direktben is meg lehet hívni a generate() metódusával. Működés szempontjából a generate metódus új ID-t generál az entitynek amennyiben az ID értéke még nincs beállítva (null), ellenkező esetben a meglévő ID-t fogja használni.
+
+Az algoritmus egy fix 16 hosszú azonosítót fog generálni [0-9a-zA-Z] random karaktereket használva, figyelembe véve az aktuális időpont nanoszekundum részét, pl. '2ZJMG008YRR4E5NW'
```

Comment:
mondjuk nem kimondottan fix 16, mert lehet rövidebb és idővel hoszabb is

Reply:
mikor lehet rovidebb vagy hosszabb? kodbol azt latom h paddolva van es fix hosszusagu karakterszamokkal dolgozik

Reply:
rövidebb megeshet bármikor, ha olyan időpontra, random, vagy sequenciára esik, aminek a String átkódolási eredménye rövidebb karakter, hosszabb még elvileg az idő tényezője miatt. De az is rémlik mintha ezt már kiküszöböltük volna...